codex/manifest/manifest.nim

```diff
@@ -214,7 +232,10 @@ func new*(
   version: CidVersion = CIDv1,
   hcodec = Sha256HashCodec,
   codec = BlockCodec,
-  protected = false): Manifest =
+  protected = false,
+  filename = "",
+  mimetype = "",
+  createdAt = now().utc.toTime.toUnix): Manifest =
```

Comment:
Okay so for the default mime type `application/octet-stream`  I think it makes sense regarding the description: 

> application/octet-stream is the default value for all other cases. An unknown file type should use this type. Browsers are particularly careful when manipulating these files to protect users from software vulnerabilities and possible dangerous behavior. 

However, for the date, I don't think we should use the beginning of the Unix epoch (01/01/1970) as a reference point. When you receive data with a zero value, it clearly indicates a non-value, unlike the default data you suggested.

As for the ?types, in my initial approach, I tried it, but after struggling a bit with Nim, I gave up and didn't revisit that approach. But I can modify the PR to use it, I guess.

Reply:
Wait, but `now().utc.toTime.toUnix` simply gives you the current timestamp? I feel this is even less likely to be interpreted as a bogus/default timestamp?

Reply:
> Wait, but now().utc.toTime.toUnix simply gives you the current timestamp? I feel this is even less likely to be interpreted as a bogus/default timestamp? 

Yes you are right. That's why I think it would be better to have those fields optional, as you mentioned: 

> I'd also investigate if we can't make these fields truly optional (i.e. ?types).

I updated this PR to include optional fields, I believe this approach makes more sense than using defaults.